pkg/kubelet/kubelet.go

```diff
@@ -479,7 +480,17 @@ func NewMainKubelet(kubeCfg *kubeletconfiginternal.KubeletConfiguration,
 
 	oomWatcher, err := oomwatcher.NewWatcher(kubeDeps.Recorder)
 	if err != nil {
-		return nil, err
+		if !libcontainersystem.RunningInUserNS() {
+			return nil, err
+		}
+		// oomwatcher.NewWatcher returns "open /dev/kmsg: operation not permitted" error,
+		// when running with sysctl value `kernel.dmesg_restrict=1`.
+		if !utilfeature.DefaultFeatureGate.Enabled(features.KubeletInUserNamespace) {
+			klog.ErrorS(err, "Failed to create an oomWatcher (running in UserNS, Hint: enable KubeletInUserNamespace feature flag to ignore the error)")
+			return nil, err
+		}
+		klog.V(2).InfoS("Failed to create an oomWatcher (running in UserNS, ignoring)", "err", err)
+		oomWatcher = nil
```

Comment:
This whole change should go behind a feature flag.

Reply:
updated